src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPathExtensions.cs

```diff
@@ -55,46 +55,55 @@ public static bool IsCollection(this ODataPath path)
             return path.LastSegment.TranslateWith(new IsCollectionTranslator());
         }
 
+        /// <summary>
+        /// Creates a new ODataPath with the specified segment added.
+        /// </summary>
+        /// <param name="path">The path against which the segment should be applied.</param>
+        /// <param name="segment">The segment applied to the path.</param>
+        /// <returns>A new ODataPath with the segment appended.</returns>
+        public static ODataPath AddSegment(this ODataPath path, ODataPathSegment segment)
+        {
+            var newPath = new ODataPath(path);

```

Comment:
> var newPath = new ODataPath(path); [](start = 11, length = 35)

It's not a big deal. The methods in this class don't check the nullability of the arguments?

Reply:
The class is internal, so an assert (if anything) would probably be more appropriate.

---
In reply to: [543923833](https://github.com/OData/odata.net/pull/1930#discussion_r543923833) [](ancestors = 543923833)